plugin-server/src/utils/internal-metrics.ts

```diff
@@ -1,9 +1,36 @@
 import Piscina from '@posthog/piscina'
-import { PluginEvent } from '@posthog/plugin-scaffold'
+import { PluginEvent, Properties } from '@posthog/plugin-scaffold'
 
 import { Hub, TeamId } from '../types'
 import { UUIDT } from './utils'
 
+export const TRACING_KEY = '$$internal_tracing' // TODO some global setting to share with py
+
+export function traceInternalProperties(properties: Properties | undefined, key: string, value: any) {
+    if (properties?.[TRACING_KEY]) {
+        properties[TRACING_KEY]['plugin_server_' + key] = value
+    }
+}
+
+export function traceInternal(event: PluginEvent, key: string, value: any) {
+    traceInternalProperties(event.properties, key, value)
+}
+
+export function traceTimingStatsd(
+    statsd: any | undefined,
+    properties: Properties | undefined,
+    startTimeKey: string,
+    key: string
+) {
+    const ingestionStartts = properties?.[TRACING_KEY]?.[startTimeKey]
+    if (ingestionStartts) {
+        const startTime = new Date(String(ingestionStartts))
+        traceInternalProperties(properties, key, new Date().getTime() - startTime.getTime())
+        properties['$' + key] = new Date().getTime() - startTime.getTime()
```

Comment:
this is not particularly useful to have on random events. It would be useful on a specific metrics event, but given we can't run queries on "all events" in PostHog, this doesn't help much.

Well, might help direct db queries. The internal metrics stuff Karl mentioned should help here